iothub/device/src/InternalClient.cs

```diff
@@ -1102,6 +1103,15 @@ public async Task SetDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCal
         {
             // Codes_SRS_DEVICECLIENT_18_003: `SetDesiredPropertyUpdateCallbackAsync` shall call the transport to register for PATCHes on it's first call.
             // Codes_SRS_DEVICECLIENT_18_004: `SetDesiredPropertyUpdateCallbackAsync` shall not call the transport to register for PATCHes on subsequent calls
+
+            // Wait to acquire the _twinSemaphore. This ensures that concurrently invoked SetDesiredPropertyUpdateCallbackAsync calls are invoked in a thread-safe manner.
+            if (Logging.IsEnabled)
```

Comment:
based on @abhipsaMisra  latest PR, you don't need to do this. Not sure if she merged it in yet or not.

Reply:
Ok sure, wasn't aware of that.

Reply:
Is it merged yet @abhipsaMisra ?

Reply:
yeah, it is in already